src/graphql/entries.ts

```diff
@@ -131,8 +148,11 @@ async function updateReferenceEntryField(
 
   // TODO: kind of duplication with line 46, check if we can combine them
   for (const key in reference.fields) {
-    const value = reference.fields[key as keyof typeof reference.fields][locale];
+    if (!isRequestedField(key, typeName, merged, gqlParams)) {
```

Comment:
TODO: check if this works without a provided query (or if merged needs to be reference, but this breaks the function again)